### PR TITLE
Update WinRM to address connect issue.

### PR DESCRIPTION
**Pull request by boumenot**
This PR is in reference to https://github.com/masterzen/winrm/pull/47.

  When the windows firewall is on, and winrm is not blocked, the
  connection will be accepted but no response will be sent by the winrm
  server.  In this case, the winrm client will wait almost indefinitely.
  This prevents packer from being able to detect when winrm is ready on a
  prepared VM.

### CODE REVIEW

**Comment by boumenot** [comment]
@masterzen Would you please review this PR to make sure I did it right, and did not miss anything.

File: vendor/github.com/packer-community/winrmcp/winrmcp/winrmcp.go [hunk 1]
```diff
@@ -42,7 +43,17 @@ func New(addr string, config *Config) (*Winrmcp, error) {
 		config = &Config{}
 	}
 
-	params := winrm.DefaultParameters()
+	connectTimeout := winrm.DefaultParameters.Timeout
+	if config.ConnectTimeout.Seconds() > 0 {
+		connectTimeout = iso8601.FormatDuration(config.ConnectTimeout)
```
Comment (masterzen) [inline vendor/github.com/packer-community/winrmcp/winrmcp/winrmcp.go]: The `Timeout` you get from the `DefaultParameters` is not of the same nature of the timeout I added. The first one is used by the winrm server, the latter is for the client side when the server doesn't answer.
I'm not sure you can safely inherit the former from the latter.

If you want winrmcp to use this new connect timeout, you need to inject it in the `Endpoint` (a bit above in the code), not in the parameters.

**Comment by masterzen** [comment]
@boumenot beside my comment on `winrmcp` usage of the connect timeout, this looks good to me.

File: vendor/github.com/packer-community/winrmcp/winrmcp/winrmcp.go (hunk 1 above)
Comment (boumenot) [inline vendor/github.com/packer-community/winrmcp/winrmcp/winrmcp.go]: Thank you!  I knew I was going to confuse the timeouts somewhere.  This change is in the vendor directory, so I need to submit it up the packer-community repo for inclusion.  I will do that.

**Comment by boumenot** [comment]
I submitted a [PR](https://github.com/packer-community/winrmcp/pull/15) to the WinRMCP repository. When that's accepted, I'll update this PR with the latest godep.

**Comment by masterzen** [comment]
@boumenot do not merge from masterzen/winrm yet, we're going to tackle masterzen/winrm#50 first.
I'm also trying to fix #2243 at the source also.

**Comment by boumenot** [comment]
OK.

**Comment by masterzen** [comment]
I've merged the latest bits in masterzen/winrm.
Note that we split the command-line tool from the library, which makes that the package name changed to `github.com/masterzen/winrm` instead of `github.com/masterzen/winrm/winrm`. This PR will have to be adapted to the new package name.

Thanks!

**Comment by cbednarski** [comment]
Thanks for the update, @masterzen!

**Comment by mwieczorek** [comment]
Hi,
I think something is wrong with this PR. 
When I have provisioner with powershell script which is running more than 60s and the script doesn't put anything to console (f.e. write-host) I got an error:
```Script exited with non-zero exit status: 1```
even if the script doesn't end/throw and exception.

I've done little investigation:
I build packer from this PR. Then run [windows_2016_winrm1.json](https://github.com/mwieczorek/packer-windows/blob/winrm/windows_2016_winrm1.json) (those are just some test templates to reproduce the issue). Provisioner has only one script with endless loop (sleeping every 30 seconds).
And in log I get:
```
2016/05/28 08:34:25 [INFO] starting remote command: powershell (...)
2016/05/28 08:35:25 packer.exe: 2016/05/28 08:35:25 [INFO] command 'powershell "& { $env:PACKER_BUILDER_TYPE=\"virtualbox-iso\"; $env:PACKER_BUILD_NAME=\"virtualbox-iso\"; c:/Windows/Temp/script.ps1; exit $LastExitCode}"' exited with code: 1
```
So it's exactly after one minute.

If I switch to master branch (or run v0.10.1) and build the same template (and the same script) - everything is working fine - packer is waiting, I don't get 'exit code 1' after a minute

Next I changed only little bit the script.
I build template [windows_2016_winrm2.json](https://github.com/mwieczorek/packer-windows/blob/winrm/windows_2016_winrm2.json) using packer built from this PR:
the only difference is that the script in the endless loop write something to console every 30 seconds (write-host). And it's working. I don't get 'exit code 1' after a minute.

Could you please check it?

**Comment by masterzen** [comment]
@mwieczorek , thanks for the report.

The symptoms you describe really look like masterzen/winrm#48 which has been fixed by now (see PR masterzen/winrm#50)

I think this present PR has to be refreshed to track the latest masterzen/winrm master.

Can you try to refresh it locally and perform again your test?
(note that upon refresh you'll probably have to rename the package imports since we changed it in the winrm project to the simpler `github.com/masterzen/winrm`.)

**Pull request closed without merging**